connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -291,6 +293,35 @@ public String configureConnector(String connName, Map<String, String> connConfig
                 "Could not execute PUT request. Error response: " + responseToString(response));
     }
 
+    /**
+     * Validate a given connector configuration. If the configuration validates or
+     * has a configuration error, an instance of {@link Response} is returned. If the validation fails
+     * an exception is thrown.
+     *
+     * @param connClassName the name of the connector class
+     * @param connConfig    the intended configuration
+     * @throws JsonProcessingException if the response failed to deserialize into a {@link ConfigInfos} instance
+     * @throws ConnectRestException if the REST api returns error status
+     * @throws ConnectException if the configuration fails to be serialized or if the request could not be sent
+     */
+    public ConfigInfos validateConnectorConfig(String connClassName, Map<String, String> connConfig)
+        throws JsonProcessingException {
+        String url = endpointForResource(String.format("connector-plugins/%s/config/validate", connClassName));
+        ObjectMapper mapper = new ObjectMapper();
+        String content;
+        try {
+            content = mapper.writeValueAsString(connConfig);
+        } catch (IOException e) {
+            throw new ConnectException("Could not serialize connector configuration and execute PUT request");
+        }
+        Response response = requestPut(url, content);
+        if (response.getStatus() < Response.Status.BAD_REQUEST.getStatusCode()) {
+            return mapper.readValue(response.getEntity().toString(), ConfigInfos.class);
```

Comment:
there's `responseToString` below, which increases the similarities with `configureConnector`